### PR TITLE
FHAC-596 HCID not populated with Prefix for AuditSourceIdentification-AuditSourceID

### PR DESCRIPTION
**Pull request by vdmehta06**
HCID is now populated with prefix AuditSourceIdentification->AuditSourceID field across all services.

### CODE REVIEW

File: Product/Production/Services/AuditRepositoryCore/src/test/java/gov/hhs/fha/nhinc/audit/transform/AuditTransformsTest.java [hunk 1]
```diff
@@ -219,6 +220,17 @@ protected void testGetActiveParticipantDestination(LogEventRequestType request,
             webContextProperties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS));
     }
 
+    protected void testAuditSourceIdentification(List<AuditSourceIdentificationType> auditSourceIdentification, AssertionType assertion) {
+        if (assertion != null && assertion.getUserInfo() != null && assertion.getUserInfo().getOrg() != null
+            && assertion.getUserInfo().getOrg().getHomeCommunityId() != null
+            && auditSourceIdentification != null && !auditSourceIdentification.isEmpty()
+            && auditSourceIdentification.size() > 0) {
```
Comment (christophermay07) [inline Product/Production/Services/AuditRepositoryCore/src/test/java/gov/hhs/fha/nhinc/audit/transform/AuditTransformsTest.java]: Two things:
1. isEmpty() and .size() > 0 is redundant (and always prefer .isEmpty())
2. isEmpty() isn't needed here; the enhanced for loop will gracefully take care of the "empty list" case.

File: Product/Production/Services/AuditRepositoryCore/src/test/java/gov/hhs/fha/nhinc/audit/transform/AuditTransformsTest.java [hunk 2]
```diff
@@ -219,6 +220,17 @@ protected void testGetActiveParticipantDestination(LogEventRequestType request,
...
+            for (AuditSourceIdentificationType auditSourceId : auditSourceIdentification) {
+                assertEquals(AuditTransformsConstants.HOME_COMMUNITY_ID_PREFIX + assertion.getUserInfo().getOrg().getHomeCommunityId(), auditSourceId.getAuditSourceID());
```
Comment (christophermay07) [inline Product/Production/Services/AuditRepositoryCore/src/test/java/gov/hhs/fha/nhinc/audit/transform/AuditTransformsTest.java]: Please wrap this line; also check the PR for other lines that are cut off (look for files that have a horizontal scroll bar on GitHub)

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java [hunk 3]
```diff
@@ -480,10 +480,12 @@ private static AuditSourceIdentificationType createAuditSourceIdentification(Str
 
         // Set the Audit Source Id (community id)
         if (communityId != null) {
-            if (communityId.startsWith("urn:oid:")) {
-                auditSrcId.setAuditSourceID(communityId.substring(8));
-            } else {
+            if (communityId.startsWith(AuditTransformsConstants.HOME_COMMUNITY_ID_PREFIX)) {
```
Comment (TabassumJafri) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java]: We can use HomeCommunityMap.getHomeCommunityIdWithPrefix(String communityId) function for getting the HCID with prefix.

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/AuditTransformsConstants.java [hunk 4]
```diff
@@ -43,4 +43,6 @@
     public static final String ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME = "DCM";
     public static final Short NETWORK_ACCESSOR_PT_TYPE_CODE_NAME = 1;
     public static final Short NETWORK_ACCESSOR_PT_TYPE_CODE_IP = 2;
+    public static final String HOME_COMMUNITY_ID_PREFIX = "urn:oid:";
```
Comment (alameluchidambaram) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/AuditTransformsConstants.java]: This constant is available in NhincConstants in CONNECTCoreLib

File: Product/SoapUI_Test/RegressionSuite/Passthru/AuditLogging-Passthrough/COREX12BatchSubmitRequest.properties [hunk 5]
```diff
@@ -52,7 +52,7 @@ ActiveParticipant_Dest.RoleIDCode.@codeSystemName=DCM
 ActiveParticipant_Dest.RoleIDCode.@displayName=Destination
 
 AuditSourceIdentification.@AuditEnterpriseSiteID=Gateway 1
-AuditSourceIdentification.@AuditSourceID=1.1
+AuditSourceIdentification.@AuditSourceID=urn:oid:1.1
```
Comment (christophermay07) [inline Product/SoapUI_Test/RegressionSuite/Passthru/AuditLogging-Passthrough/COREX12BatchSubmitRequest.properties:55]: Are we using this value as part of the request, or as part of an assertion later?

Comment (vdmehta06) [inline Product/SoapUI_Test/RegressionSuite/Passthru/AuditLogging-Passthrough/COREX12BatchSubmitRequest.properties:55]: This property is used for assertion in verifying the blob message

**Comment by christophermay07** [comment]
I've reviewed this PR, endorsing for merge without further comment.

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java [hunk 6]
```diff
@@ -480,11 +480,9 @@ private static AuditSourceIdentificationType createAuditSourceIdentification(Str
 
         // Set the Audit Source Id (community id)
         if (communityId != null) {
-            if (communityId.startsWith("urn:oid:")) {
-                auditSrcId.setAuditSourceID(communityId.substring(8));
-            } else {
-                auditSrcId.setAuditSourceID(communityId);
-            }
+            /* HomeCommunityMap.getHomeCommunityIdWithPrefix(communityId) mehtod
+             checks & takes care of prefix if not populated*/
```
Comment (TabassumJafri) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java]: you can remove the commented out code.

**Comment by TabassumJafri** [comment]
Reviewed PR and good with merge

**Comment by alameluchidambaram** [comment]
Reviewed PR and good with merge when merge conflicts are resolved.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/HomeCommunityMap.java [hunk 7]
```diff
@@ -149,8 +148,9 @@ public static String getCommunityIdFromAssertion(AssertionType assertion) {
     public static String getHomeCommunityIdFromAssertion(AssertionType assertion) {
         String homeCommunity = null;
         try {
-            if (NullChecker.isNotNullish(assertion.getHomeCommunity().getHomeCommunityId()))
+            if (NullChecker.isNotNullish(assertion.getHomeCommunity().getHomeCommunityId())) {
                 homeCommunity = assertion.getHomeCommunity().getHomeCommunityId();
+            }
         } catch (NullPointerException ex) {
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/HomeCommunityMap.java]: Remove the NullPointerException catch statement and rewrite with a Null check.

**Comment by TabassumJafri** [comment]
PR looks good and ok with merge

**Comment by alameluchidambaram** [comment]
Reviewed PR and good with merge.

**Pull request merged**